django/test/client.py

```diff
@@ -652,12 +653,20 @@ def _parse_json(self, response, **extra):
             response._json = json.loads(response.content.decode(), **extra)
         return response._json
 
-    def _handle_redirects(self, response, **extra):
+    def _handle_redirects(self, response, data='', content_type='', **extra):
```

Comment:
Default for `content_type` should be `None`? What about `data` - should that also be `None`?

Reply:
Some test methods have `None` as a default (get, post, head), others have `''` (options, put, patch, trace, delete). No sure which ones to use, I changed them to None and some tests failed. I feel like `''` is safer.

Reply:
This exchange is a little bit curious (coming to review). 

I try with `None` as default for both parameters. I get two failures in `test_follow_307_and_308_redirect`, both with the same trace ending:

```
  File "../django/django/core/handlers/wsgi.py", line 83, in __init__
    self.content_type, self.content_params = cgi.parse_header(environ.get('CONTENT_TYPE', ''))
  File "../lib/python3.6/cgi.py", line 319, in parse_header
    parts = _parseparam(';' + line)
TypeError: must be str, not NoneType
```

So it has to be `''` when we get right down to it. 

Plus we're passing `**extras` around in most methods, and there's no real place where we're checking that. `_base_environ` may be a candidate, but that's way out of scope for this ticket.